OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -92,11 +92,17 @@ public override bool Tick(Actor self)
 
 			bool targetIsHiddenActor;
 			target = RecalculateTarget(self, out targetIsHiddenActor);
+			var activeAttackTraits = attackTraits.Where(x => !x.IsTraitDisabled);
```

Comment:
I wonder if we want to `ToArray` or `ToList` this to save enumerations?

Reply:
I suspect that would actually be worse, but you'd need to profile to know for sure.

Reply:
We can avoid allocations here by following the pattern we use in other places: change the `attackTraits` definition to `readonly IEnumerable<AttackFrontal> attackTraits;` and initialize it as `attackTraits = self.TraitsImplementing<AttackFrontal>().ToArray().Where(Exts.IsTraitEnabled);`.

Reply:
Good idea. Thanks!